components/containers/core/mojom/containers.mojom

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+module containers.mojom;
+
+// Represents a container that holds a separate browsing context/identity.
+// Containers allow users to maintain different identities within the same
+// browser profile, keeping cookies, storage, and other site data separate
+// between containers.
+struct Container {
+  // Unique identifier for the container.
+  string id;
+  // User-friendly name for the container.
+  string name;
+};
+
+// Factory interface to create handlers for the containers settings page. This
+// interface is implemented on the browser side and called by the renderer to
+// establish communication channels for container management.
+interface SettingsPageHandlerFactory {
+  // Creates a new settings page handler and establishes bidirectional
+  // communication between the browser and renderer processes.
+  CreateSettingsPageHandler(
+      pending_remote<SettingsPage> page,
+      pending_receiver<SettingsPageHandler> page_handler);
+};
```

Comment:
We probably don't need this since the main interface doesn't need any parameters relevant to the page to be created

Reply:
In our own WebUIs we generally tend towards more descriptive names but I agree with @goodov - this shouldn't be blocking here, its what upstream does for all their UIs

Reply:
Well, I agree here it doesn't look this critical, so I went with a simpler approach and removed factory in favor of async `BindUI` call from JS side. We should be fine.